src/main/daml/Daml/Finance/Settlement/Batch.daml

```diff
@@ -100,7 +96,17 @@ template Batch
               Instruction.Cancel with actors
         allMustAuthorize requestors
         -- cancel instructions
-        catOptionals <$> mapA (cancelInstruction . buildKey . snd) routedStepsWithInstructionId
+        catOptionals <$> mapA (cancelInstruction . buildKey this . snd) routedStepsWithInstructionId
+
+routedSteps : Batch -> [RoutedStep]
+routedSteps Batch {routedStepsWithInstructionId} = fst <$> routedStepsWithInstructionId
```

Comment:
We could rewrite this function taking `routedStepsWithInstructionId `as input rather than the `batch`. Same for `instructionIds`.

Reply:
done in a new commit :)